modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployResultManager.java

```diff
@@ -201,13 +201,23 @@ public void updateServiceOrderTaskWithDeployResult(DeployResult deployResult,
         }
         ServiceOrderEntity entityToUpdate = new ServiceOrderEntity();
         BeanUtils.copyProperties(storedEntity, entityToUpdate);
+        DeployerTaskStatus deployerTaskStatus = deployResult.getState();
         if (deployResult.getIsTaskSuccessful()) {
-            entityToUpdate.setTaskStatus(TaskStatus.SUCCESSFUL);
+            // When the status is rollback_success, the deployment order status should be failed.
+            if (deployerTaskStatus == DeployerTaskStatus.ROLLBACK_SUCCESS) {
+                entityToUpdate.setTaskStatus(TaskStatus.FAILED);
+            } else {
+                entityToUpdate.setTaskStatus(TaskStatus.SUCCESSFUL);
+            }
             entityToUpdate.setCompletedTime(OffsetDateTime.now());
         } else {
-            entityToUpdate.setTaskStatus(TaskStatus.FAILED);
             entityToUpdate.setErrorMsg(deployResult.getMessage());
-            entityToUpdate.setCompletedTime(OffsetDateTime.now());
+            // When the status is deploy_failed, the order status should be in-progress util the
+            // rollback is done.
+            if (deployerTaskStatus != DeployerTaskStatus.DEPLOY_FAILED) {
```

Comment:
Does this code run even run while the rollback is still in-progress?

Reply:
Yes. The old code will set failed status first when deploy callback and then overwrite status to successful when rollback successful.